GitSearch/Networking/Encoding/URLEncoder.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  URLEncoder.swift
+//  GitSearch
+//
+//  Created by Vldmr on 8/26/19.
+//  Copyright © 2019 mudritskiy. All rights reserved.
+//
+
+import Foundation
+
+public struct URLEncoder {
+    
+    static func encodeParameters(for urlRequest: inout URLRequest, with parameters: HTTPParameters) throws {
+        guard let url = urlRequest.url else { throw HTTPNetworkError.missingURL }
+        
+        if var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false), let unwrappedParameters = parameters, !unwrappedParameters.isEmpty {
+            urlComponents.queryItems = [URLQueryItem]()
+            
+            for (key,value) in unwrappedParameters {
+                let queryItem = URLQueryItem(name: key, value: "\(value)")
+                
+                urlComponents.queryItems?.append(queryItem)
+            }
+            
+            urlRequest.url = urlComponents.url
+        }
+        
+    }
+    
+    static func setHeaders(for urlRequest: inout URLRequest, with headers: HTTPHeaders) throws {
+        if let headersUnwrapped = headers {
```

Comment:
`guard` instead of `if` in this context might work better. 
Also, usual pattern is to use the same name:
`guard let headers = headers else { return }`

Reply:
early exit with `guard` is useful when `if` take in a big part of code inside.
but even like that - I prefer to use `if`, because it evidently shows what to do when statement is fulfilled.
maybe I need to read more about differences `guard/if`